packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts

```diff
@@ -488,11 +488,64 @@ export interface IColumnReorderOptions {
    */
   frozenColumnCountFromEnd?: number;
 
+  /**
+   * Callback to handle the column dragstart
+   * draggedStarted indicates that the column drag has been started on DetailsHeader
+   */
+  onColumnDragStart?: (dragStarted: boolean) => void;
+
+  /**
+   * Callback to handle the column reorder
+   * draggedIndex is the source column index, that need to be placed in targetIndex
+   * Use oncolumnDrop instead of this
+   * @deprecated
+   */
+  handleColumnReorder?: (draggedIndex: number, targetIndex: number) => void;
+
   /**
    * Callback to handle the column reorder
    * draggedIndex is the source column index, that need to be placed in targetIndex
    */
-  handleColumnReorder: (draggedIndex: number, targetIndex: number) => void;
+  onColumnDrop?: (dragDropDetails: IColumnDragDropDetails) => void;
+
+  /**
+   * Callback to handle the column reorder
+   */
+  onColumnDragEnd?: (columnDropLocationDetails: ColumnDragEndLocation) => void;
+}
+
+export interface IColumnDragDropDetails {
+  /**
+   * Specifies the source column index
+   * @default -1
+   */
+  draggedIndex: number;
+
+  /**
+   * Specifies the target column index
+   * @default -1
+   */
+  targetIndex: number;
+}
+
+/**
+ * Enum to describe where the column has been dropped, after starting the drag
+ */
+export enum ColumnDragEndLocation {
```

Comment:
`header`, `surface`, and `outside` would be better names. Also, make `outside = 0` to make it so you can use `if` on it for a quick check.